modules/objdetect/include/opencv2/objdetect/aruco_board.hpp

```diff
@@ -0,0 +1,301 @@
+     */
+    CV_WRAP void setIds(InputArray ids);
+
+    /** @brief change id for ids[index]
+     * @param index - element index in ids
+     * @param newId - new value for ids[index], should be less than Dictionary size
+     */
+    CV_WRAP void changeId(int index, int newId);
+    /** @brief return ids
+     */
+    CV_WRAP const std::vector<int>& getIds() const;
+
+    /** @brief set dictionary
+     */
+    CV_WRAP void setDictionary(const Ptr<Dictionary> &dictionary);
+
+    /** @brief return dictionary
+     */
+    CV_WRAP Ptr<Dictionary> getDictionary() const;
+
+    /** @brief set objPoints
+     */
+    CV_WRAP void setObjPoints(const std::vector<std::vector<Point3f> > &objPoints);
+
+    /** @brief get objPoints
+     */
+    CV_WRAP const std::vector<std::vector<Point3f> >& getObjPoints() const;
+
+    /** @brief get rightBottomBorder
+     */
+    CV_WRAP const Point3f& getRightBottomBorder() const;
+
+    /**
+     * @brief Given a board configuration and a set of detected markers, returns the corresponding
+     * image points and object points to call solvePnP
+     *
+     * @param detectedCorners List of detected marker corners of the board.
+     * @param detectedIds List of identifiers for each marker.
+     * @param objPoints Vector of vectors of board marker points in the board coordinate space.
+     * @param imgPoints Vector of vectors of the projections of board marker corner points.
+    */
+    CV_WRAP void matchImagePoints(InputArrayOfArrays detectedCorners, InputArray detectedIds,
+                                  OutputArray objPoints, OutputArray imgPoints) const;
+
+    virtual ~Board() = default;
+protected:
+    /** @brief array of object points of all the marker corners in the board each marker include its 4 corners in this order:
+     * -   objPoints[i][0] - left-top point of i-th marker
+     * -   objPoints[i][1] - right-top point of i-th marker
+     * -   objPoints[i][2] - right-bottom point of i-th marker
+     * -   objPoints[i][3] - left-bottom point of i-th marker
+     *
+     * Markers are placed in a certain order - row by row, left to right in every row.
+     * For M markers, the size is Mx4.
+     */
+    CV_PROP std::vector<std::vector<Point3f> > objPoints;
+
+    /// the dictionary of markers employed for this board
+    CV_PROP Ptr<Dictionary> dictionary;
+
+    /// coordinate of the bottom right corner of the board, is set when calling the function create()
+    CV_PROP Point3f rightBottomBorder;
+
+    /** @brief vector of the identifiers of the markers in the board (same size than objPoints)
+     * The identifiers refers to the board dictionary
+     */
+    CV_PROP_RW std::vector<int> ids;
+};
+
+/**
+ * @brief Planar board with grid arrangement of markers
+ * More common type of board. All markers are placed in the same plane in a grid arrangement.
+ * The board can be drawn using draw() method
+ */
+class CV_EXPORTS_W GridBoard : public Board {
+public:
+    CV_WRAP GridBoard();
```

Comment:
We have `.create()` static method below.
Why do we really need public default ctor here? How it could be used?

Reply:
constructor moves to protected